dev/fattest.simplicity/src/componenttest/topology/utils/MvnUtils.java

```diff
@@ -70,296 +69,365 @@
+     * @return a list of strings that can be added to a ProcessRunner command
+     * @throws Exception
      */
-    private static String[] getJarCliProperties(LibertyServer server, List<String> nonVersionedJars) {
-
-        Map<String, String> actualJarFiles = resolveJarPaths(nonVersionedJars, server);
-        String[] addon = new String[] {};
-
-        Set<String> jarsSet = actualJarFiles.keySet();
-        for (Iterator<String> iterator = jarsSet.iterator(); iterator.hasNext();) {
-            String jarKey = iterator.next();
-            String jarPathName = actualJarFiles.get(jarKey);
-            addon = concatStringArray(addon, new String[] { "-D" + jarKey + "=" + jarPathName });
+    private ArrayList<String> getJarCliProperties() throws Exception {
+        Map<String, String> actualJarFiles = resolveJarPaths();
+        ArrayList<String> addon = new ArrayList<>();
+
+        for (Entry<String, String> entry : actualJarFiles.entrySet()) {
+            String jarKey = entry.getKey();
+            String jarPathName = entry.getValue();
+            addon.add("-D" + jarKey + "=" + jarPathName);
         }
         return addon;
     }
 
     /**
-     * Smashes two String arrays into a returned third. Useful for appending to the CLI
-     * used for ProcessBuilder. This is not a very efficient way to do this but it is
-     * easy to read/write and it is not a performance sensitive use case.
+     * Generate the array of Strings which will be used to run the "mvn clean test" command with all the appropriate parameters
      *
-     * @param a
-     * @param b
-     * @return cat a b
+     * @return an array of Strings representing the command to be run
+     * @throws Exception thrown if there was a problem assembling the parameters to the mvn command
      */
-    private static String[] concatStringArray(String[] a, String[] b) {
-        if (a == null && b != null) {
-            return Arrays.copyOf(b, b.length);
-        } else if (a != null && b == null) {
-            return Arrays.copyOf(a, a.length);
-        } else if (a == null && b == null) {
-            return new String[] {};
-        } else {
-            String[] result = new String[a.length + b.length];
-            System.arraycopy(a, 0, result, 0, a.length);
-            System.arraycopy(b, 0, result, a.length, b.length);
-            return result;
+    private String[] getMvnTestCommandArray() throws Exception {
+        String mvn = getMvn();
+
+        ArrayList<String> stringArrayList = new ArrayList<>();
+        stringArrayList.add(mvn);
+        stringArrayList.add(MVN_CLEAN); //TODO do we always want to clean?
+        stringArrayList.add(MVN_TEST);
+        stringArrayList.add("-Dwlp=" + getWLPInstallRoot());
+        stringArrayList.add("-Dtck_server=" + getServerName());
+        stringArrayList.add("-Dtck_failSafeUndeployment=" + DEFAULT_FAILSAFE_UNDEPLOYMENT);
+        stringArrayList.add("-Dtck_appDeployTimeout=" + DEFAULT_APP_DEPLOY_TIMEOUT);
```

Comment:
If there's no way to override these values, they shouldn't be named `DEFAULT_`

Reply:
pedant ;-)

Reply:
fair